aiks/canvas.cc

```diff
@@ -61,7 +61,7 @@ bool Canvas::Restore() {
 }
 
 void Canvas::Concat(const Matrix& xformation) {
-  xformation_stack_.back().xformation = xformation * GetCurrentTransformation();
+  xformation_stack_.back().xformation = GetCurrentTransformation() * xformation;
```

Comment:
Should matrix multiply use a different operator?

Reply:
I think it's consistent with glsl/glm behavior. The mult operator looks like the right order for column major, but need to double check this.

Reply:
Yup, this looks consistent with the ordering convention. Canvas transform ops are supposed to make adjustments _within_ the current transform space rather than world space (as was being assumed in some tests).

Reply:
Since the mathematical `*` operator is supposed to be commutative, using it for a matrix multiply which is non-commutative seems like a potential source of confusion. (Is that what you were getting at @zanderso?)

Reply:
In particular, why was this PR needed to switch the operands of a mathematical multiply operator?

Rhetorical answer: because it wasn't really a mathematical multiply despite the notation...

Reply:
Pre or post multiplying a matrix is not commutative. Removing the operator is fine but if we keep it matrix multiplication is just special anyway right?

Reply:
> Since the mathematical `*` operator is supposed to be commutative, using it for a matrix multiply which is non-commutative seems like a potential source of confusion. (Is that what you were getting at @zanderso?)

Yeah, that's what I was getting at. Something that makes you think about the order might help avoid issues like this. I don't have strong feelings about what the replacement would be though.

Reply:
I have a slight preference for keeping operator `*` for order dependent matrix-matrix and matrix-vector multiplication, as I think this is widely done/expected? (UE4, Unity, Godot, glsl/hlsl, [Flutter](https://api.flutter.dev/flutter/vector_math/Matrix4/operator_multiply.html), etc.)

Reply:
Hmmm. The world has already descended this slope I guess. If it's common and as long as developers keep in mind the row-vs-column notion of the matrices, then I suppose it is viable.

Although I would be more confident if the answer above was "of course this is the right order for column major matrices" rather than "let me go look this up to be sure"... ;-)